pilot/pkg/proxy/envoy/v2/ep_filters.go

```diff
@@ -85,44 +86,44 @@ func EndpointsByNetworkFilter(endpoints []endpoint.LocalityLbEndpoints, conn *Xd
 		// Iterate over all networks that have the cluster endpoint (weight>0) and
 		// for each one of those add a new endpoint that points to the network's
 		// gateway with the relevant weight
-		for n, w := range remoteEps {
-			networkConf, found := env.MeshNetworks.Networks[n]
+		for network, w := range remoteEps {
+			networkConf, found := env.MeshNetworks.Networks[network]
 			if !found {
-				adsLog.Infof("the endpoints within network %s will be ignored for no network configured", n)
+				adsLog.Infof("the endpoints within network %s will be ignored for no network configured", network)
 				continue
 			}
 			gws := networkConf.Gateways
 			if len(gws) == 0 {
-				adsLog.Infof("the endpoints within network %s will be ignored for no gateways configured", n)
+				adsLog.Infof("the endpoints within network %s will be ignored for no gateways configured", network)
 				continue
 			}
 
+			registryName := getNetworkRegistry(networkConf)
 			gwEps := []endpoint.LbEndpoint{}
 			// There may be multiples gateways for the network. Add an LbEndpoint for
 			// each one of them
 			for _, gw := range gws {
-				var gwEp *endpoint.LbEndpoint
-				//TODO add support for getting the gateway address from the service registry
-
-				// If the gateway address in the config was a hostname it got already resolved
-				// and replaced with an IP address when loading the config
-				if gwIP := net.ParseIP(gw.GetAddress()); gwIP != nil {
-					addr := util.BuildAddress(gw.GetAddress(), gw.Port)
-					gwEp = &endpoint.LbEndpoint{
-						HostIdentifier: &endpoint.LbEndpoint_Endpoint{
-							Endpoint: &endpoint.Endpoint{
-								Address: &addr,
+				gwAddresses := getGatewayAddresses(gw, registryName, env)
+				// If gateway addresses are found, create an endpoint for each one of them
+				if len(gwAddresses) > 0 {
+					for _, gwAddr := range gwAddresses {
+						epAddr := util.BuildAddress(gwAddr, gw.Port)
```

Comment:
and here you will end up sticking the ingress hostname from above. which will cause eds to crash

Reply:
Thanks, got it.